File: ports/csharp/src/network/socket/BcsTcpServer.cs
using System;
using System.Net;
using System.Net.Sockets;
using System.Threading.Tasks;

namespace BCS.Network.Socket {
    // BcsTcpServer abstracts a network listener socket for C#
    public class BcsTcpServer {
        private TcpListener _listener;

        public void Listen(int port) {
            _listener = new TcpListener(IPAddress.Any, port);
            _listener.Start();
        }

        public async Task<BcsTcpSocket> AcceptAsync() {
            var client = await _listener.AcceptTcpClientAsync();
            var socket = new BcsTcpSocket();
            // In a real port, BcsTcpSocket would accept an existing TcpClient via constructor
            return socket;
        }

        public void Close() {
            _listener?.Stop();
        }
    }
}

File: ports/java/src/network/socket/BcsTcpServer.java
package bcs.network.socket;

import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

// BcsTcpServer abstracts a network listener socket for Java
public class BcsTcpServer {
    private ServerSocket serverSocket;

    public void listen(int port) throws IOException {
        serverSocket = new ServerSocket(port);
    }

    public BcsTcpSocket accept() throws IOException {
        if (serverSocket != null) {
            Socket clientSocket = serverSocket.accept();
            BcsTcpSocket socket = new BcsTcpSocket();
            // Internal logic would set the underlying java.net.Socket
            return socket;
        }
        return null;
    }

    public void close() throws IOException {
        if (serverSocket != null && !serverSocket.isClosed()) {
            serverSocket.close();
        }
    }
}
